src/datepicker/datepicker-service.ts

```diff
@@ -12,11 +12,15 @@ import {filter} from 'rxjs/operator/filter';
 export class NgbDatepickerService {
   private _model$ = new Subject<DatepickerViewModel>();
 
+  private _dateSelected$ = new Subject<boolean>();
+
   private _state: DatepickerViewModel =
       {disabled: false, displayMonths: 1, firstDayOfWeek: 1, focusVisible: false, months: [], selectedDate: null};
 
   get model$() { return filter.call(this._model$.asObservable(), model => model.months.length > 0); }
 
+  get dateSelected$() { return filter.call(this._dateSelected$.asObservable(), selected => !!selected); }
```

Comment:
please add the following tests to the `datepicker-service.spec.ts`:

- `focus()` → should emit date regardless if it's the same one or not
- `focusSelect()` → should emit date regardless if it's the same one or not